java/operator/org.eclipse.theia.cloud.operator/src/main/java/org/eclipse/theia/cloud/operator/util/K8sUtil.java

```diff
@@ -164,6 +170,21 @@ private static <T extends HasMetadata, U, V extends Resource<T>> Optional<T> loa
                 return Optional.empty();
             }
 
+            // Apply labels to the resource metadata
+            if (newItem.getMetadata().getLabels() == null) {
+                newItem.getMetadata().setLabels(new HashMap<>());
+            }
+            newItem.getMetadata().getLabels().putAll(labelsToAdd);
+
+            // If the resource is a Deployment, also apply labels to the pod template metadata
+            if (newItem instanceof Deployment) {
+                Deployment deployment = (Deployment) newItem;
```

Comment:
Minor: As we use Java 17, you should be able to simplify this:
```suggestion
            if (newItem instanceof Deployment deployment) {
```